internal/testing/fake/mod.go

```diff
@@ -1111,3 +1114,41 @@ func (f AccessControlFactory) Deserialize(serde.Context, []byte) (serde.Message,
 func (f AccessControlFactory) AccessOf(serde.Context, []byte) (access.Access, error) {
 	return NewAccessControl(), f.err
 }
+
+// WaitLog is a helper to wait for a log to be printed. It executes the callback
+// when it detects it.
+func WaitLog(msg string, t time.Duration, cb func()) (zerolog.Logger, *bytes.Buffer) {
+	reader, writer := io.Pipe()
+	done := make(chan struct{})
+
+	buffer := new(bytes.Buffer)
+	tee := io.TeeReader(reader, buffer)
+
+	go func() {
+		select {
+		case <-done:
+		case <-time.After(t):
+			writer.Close()
+		}
+	}()
+
+	go func() {
+		defer cb()
+		defer close(done)
+
+		data := make([]byte, 1024)
```

Comment:
just hope that your message is never cut between chunks of 1024 bytes ;)

Reply:
well the log would be too long anyway in that case, but that's for testing so the value can change.